### ISSUE TITLE
error retrieving tap attribute from sysfs when migrating KVM instance

### ISSUE THREAD

**Issue by grisu48** [openSUSE/wicked#802]
Hey there,

I'm getting a really weird error, when trying to migrate a KVM instance using virsh. I have two identical machines 'aa' (aurora-australis) and 'ab' (aurora-borealis) and I am trying to migrate a VM from aa to ab by doing

    aa # virsh migrate --live --persistent --verbose --copy-storage-all vault43 qemu+ssh://ab/system
    error: internal error: unable to execute QEMU command 'drive-mirror': Failed to connect socket: No route to host

When looking into the journal of 'ab', the only apparent error is the following

    [...]
    Jul 16 17:07:02 aurora-borealis wickedd[1201]: error retrieving tap attribute from sysfs
    [...]

Full relevant log: [wickedd.txt](https://github.com/openSUSE/wicked/files/3397507/wickedd.txt)

I am running openSuSE 15.0 LEAP with all updated installed and in place. The machines have been rebooted, but the error persists.

    Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
    openSuSE 15.0 LEAP
    Linux aurora-borealis 4.12.14-lp150.12.64-default #1 SMP Mon Jun 17 16:53:50 UTC 2019 (3edfd41) x86_64 x86_64 x86_64 GNU/Linux
    Linux aurora-australis 4.12.14-lp150.12.64-default #1 SMP Mon Jun 17 16:53:50 UTC 2019 (3edfd41) x86_64 x86_64 x86_64 GNU/Linux

**Comment by rtorrero** [comment]
Hi, 

Could you provide additional details as described in https://en.opensuse.org/openSUSE:Bugreport_wicked ?

Thanks!

**Comment by grisu48** [comment]
Hi there, I collected all information according to the website and post them in the following tar.gz archive.
[wicked.tar.gz](https://github.com/openSUSE/wicked/files/3406704/wicked.tar.gz)

Greetings

**Comment by rtorrero** [comment]
Hi!,

I don't really think that it's related to wicked, but let's see:
How are ab and aa connected? do they have connectivity regardless the virsh command? Looking at this -> https://bugzilla.redhat.com/show_bug.cgi?id=1231917 may give you some clues.

**Comment by grisu48** [comment]
Hi rtorrero,

thanks for the reply. ab and aa are directly connected (static IP on both machines) and the connectivity works fine - Ping, and ssh works from fine from each machine. I also check that the ssh keys are on each machine, so that passwordless login works in both directions

    ➜  ~ ssh aa
    Last login: Fri Aug 16 22:52:10 2019 from 138.232.146.54
    aurora-australis:~ # ping ab
    PING aurora-borealis (192.168.101.101) 56(84) bytes of data.
    64 bytes from aurora-borealis (192.168.101.101): icmp_seq=1 ttl=64 time=0.157 ms
    --- aurora-borealis ping statistics ---
    2 packets transmitted, 2 received, 0% packet loss, time 1021ms
    rtt min/avg/max/mdev = 0.145/0.151/0.157/0.006 ms
    
    aurora-australis:~ # ssh ab
    Last login: Fri Aug 16 15:53:45 2019 from 138.232.146.54
    
    
    aurora-borealis:~ # ping aa
    PING aurora-australis (192.168.101.102) 56(84) bytes of data.
    64 bytes from aurora-australis (192.168.101.102): icmp_seq=1 ttl=64 time=0.103 ms
    --- aurora-australis ping statistics ---
    1 packets transmitted, 1 received, 0% packet loss, time 0ms
    rtt min/avg/max/mdev = 0.103/0.103/0.103/0.000 ms
    
    aurora-borealis:~ # ping ab
    PING aurora-borealis (192.168.101.101) 56(84) bytes of data.
    64 bytes from aurora-borealis (192.168.101.101): icmp_seq=1 ttl=64 time=0.026 ms
    --- aurora-borealis ping statistics ---
    2 packets transmitted, 2 received, 0% packet loss, time 1026ms
    rtt min/avg/max/mdev = 0.026/0.029/0.033/0.006 ms
    
    aurora-borealis:~ # ssh aa
    Last login: Sat Aug 17 00:06:18 2019 from 138.232.146.54
    aurora-australis:~ # 

I'm trying to get a test setup running to figure out, where the error originates from. Apart from the default configuration the only thing I've changed in the network setup was to setup a network bridge via YaST.

**Comment by grisu48** [comment]
I think we can close this issue. I tried to reproduce the issue in a own setup, where I created two VMs: `pion` and `muon`. Within `pion` I then created a nested VM `vault27`, installed it an run a live migration with

    virsh migrate --live --persistent --verbose --copy-storage-all vault27 qemu+ssh://muon/system

Live migration with storage transfer works from `pion` to `muon` and vice-versa nicely. Then I switched from the default network configuration to a bridged network and re-tested the procedure.
In my test setup the process works nicely without any other issues.

Since I cannot reproduce the error using a clean setup, it might be a issue on the bare metal configuration, and not in wicked or libvirt.

So, not reproducable, I'm closing the issue. Thanks for the reply!
